File: TDDMicroExercises.ProposedSolution/Javascript/unicode-file-to-htm-text-converter-tests/unicode-file-to-htm-text-converter-test.js
describe('Unicode To Htm Converter', function () {

	describe('UnicodeFileToHtmTextConverter', function () {

		it('should convert ampersand', function () {

			var stubTextStream = a.stub(TextStream);
			spyOn(stubTextStream, 'getText').andCallFake(function () { return 'Cut & Paste'; });
			var target = new UnicodeFileToHtmTextConverter(stubTextStream);

			var result = target.convertToHtml();

			expect(result).toEqual('Cut &amp; Paste');
			
		});

		it('should convert greate than and less than', function () {

			var stubTextStream = a.stub(TextStream);
			spyOn(stubTextStream, 'getText').andCallFake(function () { return '10 > 5; 5 < 10'; });
			var target = new UnicodeFileToHtmTextConverter(stubTextStream);

			var result = target.convertToHtml();

			expect(result).toEqual('10 &gt; 5; 5 &lt; 10');

		});


		it('should add breakline for multiple lines', function () {

			var stubTextStream = a.stub(TextStream);
			spyOn(stubTextStream, 'getText').andCallFake(function () { return 'hello\nhow are you doing?\n'; });
			var target = new UnicodeFileToHtmTextConverter(stubTextStream);

			var result = target.convertToHtml();

			expect(result).toEqual('hello<br />how are you doing?<br />');

		});
	});

});

File: TDDMicroExercises.ProposedSolution/CSharp/UnicodeFileToHtmTextConverter.Tests/UnicodeFileToHtmTextConverterTest.cs
using NUnit.Framework;

namespace TDDMicroExercises.OneSolution.UnicodeFileToHtmTextConverter.Tests
{
    [TestFixture]
    public class UnicodeFileToHtmTextConverterTest
    {
        [Test]
        public void should_convert_ampersand()
        {
            StubUnicodeTextSource stubSource = new StubUnicodeTextSource();
            stubSource.StubCallToGetTextReader("Cut & Paste");

            UnicodeFileToHtmTextConverter target = new UnicodeFileToHtmTextConverter(stubSource);
            string result = target.ConvertToHtml();

            Assert.AreEqual(result, "Cut &amp; Paste<br />", "html converted text");
        }

        [Test]
        public void should_convert_greate_than_and_less_than()
        {
            StubUnicodeTextSource stubSource = new StubUnicodeTextSource();
            stubSource.StubCallToGetTextReader("10 > 5; 5 < 10");

            UnicodeFileToHtmTextConverter target = new UnicodeFileToHtmTextConverter(stubSource);
            string result = target.ConvertToHtml();

            Assert.AreEqual(result, "10 &gt; 5; 5 &lt; 10<br />", "html converted text");
        }

        [Test]
        public void should_add_breakline_for_multiple_lines()
        {
            StubUnicodeTextSource stubSource = new StubUnicodeTextSource();
            stubSource.StubCallToGetTextReader("hello\nhow are you doing?");

            UnicodeFileToHtmTextConverter target = new UnicodeFileToHtmTextConverter(stubSource);
            string result = target.ConvertToHtml();

            Assert.AreEqual(result, "hello</b>how are you doing?<br />", "html converted text");
        }

    }
}
